### PR TITLE
[chore] add warning documentation around healthcheck extension

### PR DESCRIPTION
**Pull request by codeboten**
This follows https://github.com/open-telemetry/opentelemetry-collector/pull/9065#issuecomment-1852396742

### CODE REVIEW

**Review by TylerHelmuth — APPROVED** [review]
Should we explicitly mention this component is planned to be marked Deprecated in the future?

**Review by mx-psi — APPROVED** [review]

File: extension/healthcheckextension/README.md [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # Health Check
 
+> ⚠️⚠️⚠️ **Warning** ⚠️⚠️⚠️
+>
+> The `check_collector_pipeline` feature of this extension is not working as expected. It
+> is recommended to not use the feature. The work to to add a new version of the healthcheck extension
```
Comment (mx-psi) [inline extension/healthcheckextension/README.md:6]:
```suggestion
> is recommended to not use the feature. The work to add a new version of the healthcheck extension
```

**Comment by mx-psi** [comment]
> Should we explicitly mention this component is planned to be marked Deprecated in the future?

IMO let's deprecate it once we are closer to having a replacement, the extension is still useful in some basic cases and we don't have a replacement yet

**Comment by codeboten** [comment]
> IMO let's deprecate it once we are closer to having a replacement, the extension is still useful in some basic cases and we don't have a replacement yet

either that or the new healthcheck extension may be able to guarantee backwards compatibility with the functionality we know works for this extension (the basic case), and then we don't need to migrate anything

**Review by crobert-1 — APPROVED** [review]

**Pull request merged**